### DISCUSSION TITLE
kiali multi-cluster solution

### DISCUSSION THREAD

**Discussion by LeslieKuo** [kiali/kiali#3952] [Multicluster]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
**Solution1:** Focus on Home cluster and ONLY display the entire service topology on it. Select MeshID and then display the entire mesh which also focus on homecluster. If some virtual services spanned on multi-cluster, just display the related elements on other clusters. 
**Solution2:** Display the entire mesh topology on every cluster in details.

In my view the current multi-cluster feature is more like **Solution1**.  Seems like we install Kiali in every cluster(Not sure).  After mesh selection, only display the homecluster details. Other cluster only show the releated services. If I want see more details in other clusters, I need to  switch to another mesh. 

**Describe alternatives you've considered**
May need something like **Solution2** to support multi-cluster. Maybe the prometheus metrix collection is easier. Particularly multicluster k8s resource collection. So we can observe the entire mesh in details at once.
Also need more arch diagram and docs about multi-cluster feature. (How to deploy kiali corresponding to different Istio multicluster deployment models )

**Comment by lucasponce** [comment]
Thanks @LeslieKuo for the issue ! I've moved into a discussion as we are studying all multicluster/federated deployments model in Istio and how Kiali can fit in, so, your feedback is most than welcome. We'd like to invite you into more active discussion around that.

**Reply by jshaughn** [reply]
@LeslieKuo You are right, current Kiali MC support is focused on the Multi-Primary Istio deployment model.  The general approach is to deploy a Kiali with each Istiod and have it configured that a Prometheus is deployed for each of those control plane instances.  And so the traffic view is focused on the traffic within the "home" cluster while touching incoming and outgoing requests to services on other clusters.  With Kiali 1.33 you can now link directly to a remote Kiali instance (if your Kiali services are properly annotated).

For situations where different control planes/clusters are individually managed and may not want external exposure, this approach is appropriate.

But as for Solution2, we definitely understand that other situations call for a more unified view.  We are actively designing in this area and as @lucasponce mentions, we value any feedback.   In general I think this would require a unified Prometheus, such that different control planes were configured to scrape to the single instance.  Is this how you would approach your installation, or do you have a different architecture?
